bugsnag-android-core/src/main/java/com/bugsnag/android/ReportDeliveryDelegate.java

```diff
@@ -0,0 +1,109 @@
+package com.bugsnag.android;
+
+import androidx.annotation.NonNull;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.RejectedExecutionException;
+
+class ReportDeliveryDelegate extends BaseObservable {
+
+    final Logger logger;
+    final EventStore eventStore;
+    final ImmutableConfig immutableConfig;
+    final BreadcrumbState breadcrumbState;
+
+    ReportDeliveryDelegate(Logger logger, EventStore eventStore,
+                           ImmutableConfig immutableConfig, BreadcrumbState breadcrumbState) {
+        this.logger = logger;
+        this.eventStore = eventStore;
+        this.immutableConfig = immutableConfig;
+        this.breadcrumbState = breadcrumbState;
+    }
+
+    void deliverEvent(@NonNull Event event) {
```

Comment:
Should this be called just `deliver` and rename the non-public method `deliverReportInternal` (or similar)?

Reply:
Renamed methods